src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -48,34 +47,21 @@ public BodyModelBinder()
                 return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
             }
 
-            object model = null;
             try
             {
-                model = await formatter.ReadAsync(formatterContext);
+                var model = await formatter.ReadAsync(formatterContext);
+
+                // key is empty to ensure that the model name is not used as a prefix for validation.
+                return new ModelBindingResult(model, key: string.Empty, isModelSet: true);
             }
             catch (Exception ex)
             {
-                model = GetDefaultValueForType(bindingContext.ModelType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, ex);
 
                 // This model binder is the only handler for the Body binding source.
                 // Always tell the model binding system to skip other model binders i.e. return non-null.
                 return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
```

Comment:
* For action parameters, `MethodInfo.Invoke` seems to take care of coercing null value types to default values for value types (this is verified by test). 
* For value type properties that are annotated with `[FromBody]`, we end up getting a null ref as part of the property setter. Tracking this via https://github.com/aspnet/Mvc/issues/2357. 

That said, it seems like fixing it specifically for `BodyModelBinder` is incorrect and we should have a single top level piece of code that addresses this, not individual binders.